modal/volume.py

```diff
@@ -12,6 +13,9 @@
 class _VolumeHandle(_Handle, type_prefix="vo"):
     """mdmd:hidden Handle to a `Volume` object."""
 
+    # To prevent multiple concurrent operations on the same volume.
+    _lock: asyncio.Lock = asyncio.Lock()
```

Comment:
I think this would actually create a lock so you can use *one of any* VolumeHandles on your running task, since the assignment is in class scope (this would not create an instance-specific variable in Python as might be expected in other languages, but a class variable). 

To set an instance variable you'd normally set it in your `__init__` method, but since our _Handle's are a bit special you'd set it by overloading _initialize_from_empty, e.g. 
```python
def _initialize_from_empty(self):
    self._lock = asyncio.Lock()
```

The type annotation can remain in the class scope however

Reply:
Doh, yeah...

Reply:
Was testing a class lock for this scenario btw:

```
    stub = modal.Stub()
    stub.vol = modal.Volume()

    deploy_stub(stub, "vol", client=client)

    f = Volume.lookup("vol", client=client)
    g = Volume.lookup("vol", client=client)
    assert f is g  # nope
```